tools/PowerShell/Microsoft.WinGet.Client/src/Library/Invoke-WinGetCommand.ps1

```diff
@@ -0,0 +1,238 @@
+class WinGetSource
+{
+    [string] $Name
+    [string] $Argument
+    [string] $Data
+    [string] $Identifier
+    [string] $Type
+
+    WinGetSource ()
+    {  }
+
+    WinGetSource ([string]$a, [string]$b, [string]$c, [string]$d, [string]$e)
+    {
+        $this.Name       = $a.TrimEnd()
+        $this.Argument   = $b.TrimEnd()
+        $this.Data       = $c.TrimEnd()
+        $this.Identifier = $d.TrimEnd()
+        $this.Type       = $e.TrimEnd()
+    }
+
+    WinGetSource ([string[]]$a)
+    {
+        $this.name       = $a[0].TrimEnd()
+        $this.Argument   = $a[1].TrimEnd()
+        $this.Data       = $a[2].TrimEnd()
+        $this.Identifier = $a[3].TrimEnd()
+        $this.Type       = $a[4].TrimEnd()
+    }
+    
+    WinGetSource ([WinGetSource]$a)
+    {
+        $this.Name       = $a.Name.TrimEnd()
+        $this.Argument   = $a.Argument.TrimEnd()
+        $this.Data       = $a.Data.TrimEnd()
+        $this.Identifier = $a.Identifier.TrimEnd()
+        $this.Type       = $a.Type.TrimEnd()
+
+    }
+    
+    [WinGetSource[]] Add ([WinGetSource]$a)
+    {
+        $FirstValue  = [WinGetSource]::New($this)
+        $SecondValue = [WinGetSource]::New($a)
+        
+        [WinGetSource[]] $Combined = @([WinGetSource]::New($FirstValue), [WinGetSource]::New($SecondValue))
+
+        Return $Combined
+    }
+
+    [WinGetSource[]] Add ([String[]]$a)
+    {
+        $FirstValue  = [WinGetSource]::New($this)
+        $SecondValue = [WinGetSource]::New($a)
+        
+        [WinGetSource[]] $Combined = @([WinGetSource]::New($FirstValue), [WinGetSource]::New($SecondValue))
+
+        Return $Combined
+    }
+}
+
+class WinGetPackage
```

Comment:
I think there is a bug somewhere in the constructor for this class. Not something that needs to be fixed for this first iteration of the pre-release, but maybe something that should be tracked in an issue somewhere, as I would consider this critical for a minimally viable product

-----
# Test Case
```
PS> Get-WinGetPackage | Where-Object {$_.Available} | Format-Table
```
## Expected Result
- A list of only the packages which contained a value in `Available`
## Actual Result
- A list of all the packages
## Root Cause 
- I believe the root cause is that the constructor for this class sets the value to an empty string rather than `$null`
- Root cause can be verified using
 `Get-WinGetPackage | Where-Object {!([string]::IsNullOrWhiteSpace($_.Available))} | Format-Table`